routers/api/v1/admin/email.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package admin
+
+import (
+	"net/http"
+
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/context"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/routers/api/v1/utils"
+	"code.gitea.io/gitea/services/convert"
+)
+
+// GetAllEmails
+func GetAllEmails(ctx *context.APIContext) {
+	listOptions := utils.GetListOptions(ctx)
+
+	emails, maxResults, err := user_model.SearchEmails(&user_model.SearchEmailOptions{
+		Keyword:     ctx.Params(":email"),
+		ListOptions: listOptions,
+	})
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "GetAllEmails", err)
+		return
+	}
+
+	results := make([]*api.Email, len(emails))
+	for i := range emails {
+		results[i] = convert.ToEmailSearch(emails[i])
+	}
+
+	ctx.SetLinkHeader(int(maxResults), listOptions.PageSize)
+	ctx.SetTotalCountHeader(maxResults)
+	ctx.JSON(http.StatusOK, &results)
+}
+
+// GetEmail
+func GetEmail(ctx *context.APIContext) {
+	GetAllEmails(ctx)
```

Comment:
Is `Params(":email")` an accurate condition or inaccurate one? Maybe they should be treated differently.

Reply:
It's accurate, as from routes the `{email}` route param is available to this route:
```
m.Group("/emails", func() {
...
  m.Group("/{email}", func() {
	  m.Get("", admin.GetEmail)
  })
})
```

Reply:
But if the router is `.../my123@123.com`, and there are two emails in the databases, one is `my123@123.com` which is correctly matched, another is `mmy123@123.com` which should not be matched but will.

Reply:
ah, ok. I understand now. Thank you for details. Let me adjust this.

Reply:
I've removed this endpoint, as it has too much overlap with Search Endpoint.